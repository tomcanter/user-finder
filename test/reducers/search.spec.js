import searchReducer from '../../src/store/Search/reducer';

describe('Reducer: search', () => {

  describe('when passed no state', () => {
    it('should return the initial state', () => {
      expect(
        searchReducer(undefined, {})
      ).toMatchSnapshot();
    });
  });

  describe('when a search request is pending', () => {
    it('should set isPending to true', () => {
      const beforeState = {
        isPending: false,
      };
      const afterState = searchReducer(beforeState, {type: 'SEARCH_REQUEST'});

      expect(
        afterState
      ).toMatchSnapshot();
      expect(afterState).not.toEqual(beforeState);
    });
  });

  describe('when a search request is successful', () => {
    it('should update the state with the payload', () => {
      const beforeState = {
        error: null,
        isPending: true,
        result: [],
      };
      const action = {
        entities: {},
        type: 'SEARCH_SUCCESS',
        result: [1, 2],
      };
      const afterState = searchReducer(beforeState, action);

      expect(
        afterState
      ).toMatchSnapshot();
      expect(afterState).not.toEqual(beforeState);
    });
  });

  describe('when a search request fails', () => {
    it('should update the state with the error message', () => {
      const beforeState = {
        error: null,
        isPending: true,
        result: [],
      };
      const action = {
        response: {},
        message: 'The server died',
        type: 'SEARCH_FAILURE',
      };
      const afterState = searchReducer(beforeState, action);

      expect(
        afterState
      ).toMatchSnapshot();
      expect(afterState).not.toEqual(beforeState);
    });
  });

});
