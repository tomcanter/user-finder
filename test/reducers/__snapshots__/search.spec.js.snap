exports[`Reducer: search when a search request fails should update the state with the error message 1`] = `
Object {
  "error": Object {
    "message": "The server died",
    "response": Object {},
  },
  "isPending": false,
  "result": Array [],
}
`;

exports[`Reducer: search when a search request is pending should set isPending to true 1`] = `
Object {
  "isPending": true,
}
`;

exports[`Reducer: search when a search request is successful should update the state with the payload 1`] = `
Object {
  "error": null,
  "isPending": false,
  "result": Array [
    1,
    2,
  ],
}
`;

exports[`Reducer: search when passed no state should return the initial state 1`] = `
Object {
  "error": null,
  "isPending": false,
  "result": Array [],
}
`;
